### PR TITLE
[SNOW-837919] Fix test by adding exception handling for grant query

### PR DESCRIPTION
**Pull request by sfc-gh-yizhou**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-837919.

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

**Comment by sfc-gh-yizhou** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by sfc-gh-yizhou** [comment]
recheck

File: test/integ/test_connection.py [hunk 1]
```diff
@@ -390,11 +390,16 @@ def exe(sql):
...
+        try:
+            # This statement will be partially executed because REFERENCE_USAGE
+            # will not be granted.
+            exe(
+                "grant all on database {} to role snowdog_role".format(
+                    db_parameters["database"]
+                )
             )
-        )
+        except Exception:
```
[the remark this answers is not in the document]
Comment (sfc-gh-yizhou) [reply test/integ/test_connection.py]: Thank you for the feedback. Yes, we should be explicit about the cause of the exception. I added an assertion for the error number `003011` and an assertion for the error message.

File: DESCRIPTION.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
   - Added the ability to read Snowflake's central configuration file.
   - Improved OCSP response caching to remove tmp cache files on Windows.
   - Added a parameter `server_session_keep_alive` in `SnowflakeConnection` that skips session deletion when client connection closes.
+  - Add exception handling to `test.integ.test_connection.test_drop_create_user` when executing `grant all on database to role` as the grant is partially executed.
```
Comment (sfc-gh-stan) [inline DESCRIPTION.md:15]: I think we don't need to include this in the release notes.

Comment (sfc-gh-yizhou) [reply DESCRIPTION.md]: Thanks for adding this label for me. I have removed it.

**Review by sfc-gh-stan — APPROVED** [review]

**Review by sfc-gh-mkeller — APPROVED** [review]
Please verify on Jenkins that the test passes before merging

:ship: :ship:

**Comment by sfc-gh-yizhou** [comment]
The test `integ/test_connection::test_drop_create_user` passes in all 6 downstream jobs (from the console logs of jobs 36 - 311). Thank you all for your help.

**Pull request merged**